workitem/link/link_repository.go

```diff
@@ -216,6 +217,21 @@ func (r *GormWorkItemLinkRepository) Delete(ctx context.Context, ID string) erro
 	return nil
 }
 
+// DeleteRelatedLinks deletes all links in which the source or target equals the
+// given work item ID.
+func (r *GormWorkItemLinkRepository) DeleteRelatedLinks(ctx context.Context, wiIDStr string) error {
+	wiId, err := strconv.ParseUint(wiIDStr, 10, 64)
+	if err != nil {
+		// treat as not found: clients don't know it must be a uint64
+		return errors.NewNotFoundError("work item link", wiIDStr)
+	}
+	db := r.db.Where("? in (source_id, target_id)", wiId).Delete(&WorkItemLink{})
```

Comment:
Sure, you would have to call `Unscoped()` like so in order to hard-delete:

```go
db := r.db.Unscoped().Where("? in (source_id, target_id)", wiId).Delete(&WorkItemLink{})
```

See the [gorm documentation on delete](http://jinzhu.me/gorm/crud.html#delete).